tests/foreman/ui_airgun/test_smartvariable.py

```diff
@@ -248,3 +248,948 @@ def test_positive_create_matcher_attribute_priority(
+            module_host.name,
+            variable_name,
+            dict(value=variable_value, overridden=True)
+        )
+        host_smart_variable_value = session.host.get_puppet_class_parameter_value(
+            module_host.name, variable_name)
+        assert host_smart_variable_value['value'] == variable_value
+        assert host_smart_variable_value['overridden'] is True
+        values = session.smartvariable.search(variable_name)
+        assert values[0]['Variable'] == variable_name
+        assert values[0]['Number of Overrides'] == '1'
+
+
+@tier2
+def test_positive_override_default_value_from_attribute(session, module_host, puppet_class):
+    """Override smart variable that has default value with a new value from
+    attribute(host) page
+
+    :id: a76dcb34-b005-4998-99e9-418b2e821a00
+
+    :steps:
+
+        1.  Create a variable with array type and default value
+        2.  From host/hostgroup, override the variable value.
+        3.  Submit the changes.
+
+    :expectedresults:
+
+        1.  The host/hostgroup is saved with changes and variable value has
+            been changed
+        2.  Smart variable should be treated as overridden
+
+    :BZ: 1405118
+
+    :CaseLevel: Integration
+    """
+    variable_name = gen_string('alpha')
+    variable_default_value = '[20,30]'
+    variable_new_value = '[90,100,120]'
+    with session:
+        session.smartvariable.create({
+            'variable.key': variable_name,
+            'variable.puppet_class': puppet_class.name,
+            'variable.key_type': 'array',
+            'variable.default_value': variable_default_value,
+        })
+        values = session.smartvariable.search(variable_name)
+        assert values[0]['Variable'] == variable_name
+        assert values[0]['Number of Overrides'] == '0'
+        session.host.set_puppet_class_parameter_value(
+            module_host.name,
+            variable_name,
+            dict(value=variable_new_value, overridden=True)
+        )
+        host_smart_variable_value = session.host.get_puppet_class_parameter_value(
+            module_host.name, variable_name)
+        assert host_smart_variable_value['value'] == variable_new_value
+        assert host_smart_variable_value['overridden'] is True
+        values = session.smartvariable.search(variable_name)
+        assert values[0]['Variable'] == variable_name
+        assert values[0]['Number of Overrides'] == '1'
+
+
+@tier2
+def test_negative_create_override_from_attribute(session, module_host, puppet_class):
+    """No impact on variable on overriding the variable with invalid value
+    from attribute.
+
+    :id: 18071443-a511-49c4-9ca9-04c7594b831d
+
+    :steps:
+
+        1.  Create a variable.
+        2.  Associate variable with fqdn/hostgroup.
+        3.  From host/hostgroup, attempt to override the variable with some
+            other key type of value.
+
+    :expectedresults:
+
+        1.  Error thrown for invalid type value.
```

Comment:
I see no validation for that. Do we have pop-up there? Could you add proper assertion?

Reply:
addressed